### ISSUE TITLE
tests: kernel/mem_protect/mem_protect is failed on nucleo_f429zi/disco_l475_iot1

### ISSUE THREAD

**Issue by erwango** [zephyrproject-rtos/zephyr#7877]
Tested on e9174c6ddeddf58d8aebba5c89e50965731899e6.

3 sections failed:
```
starting test - test_mem_domain_valid_access
***** MPU FAULT *****
  Executing thread ID (thread): 0x2000790c
  Faulting instruction address:  0x8001bac
  Data Access Violation
  Address: 0x20007ba0
Caught system error -- reason 0
FAIL - test_mem_domain_valid_access
```
```
starting test - test_mem_domain_partitions_user_rw
***** MPU FAULT *****
  Executing thread ID (thread): 0x20007c40
  Faulting instruction address:  0x8001b7c
  Data Access Violation
  Address: 0x20007ba0
Caught system error -- reason 0
FAIL - test_mem_domain_partitions_user_rw
```
```
starting test - test_mem_domain_remove_partitions
***** MPU FAULT *****
  Executing thread ID (thread): 0x20007a24
  Faulting instruction address:  0x8001990
  Data Access Violation
  Address: 0x20007ba0
Caught system error -- reason 0
FAIL - test_mem_domain_remove_partitions
```

**Comment by andrewboie** [comment]
I don't have this hardware available to me but I did a build and examined the binary.

It appears that the MPU hardware is not being programmed correctly for the memory domain configuration. The tests are trying to read/write a buffer that they should have been granted access to but instead we fault.

I don't know if this is a recent regression, or a problem that has been around for a long time only exposed by recently enabling many more tests to run in user mode on non-x86.

I'd be very interested to know if this is limited to just this board/SOC or if all ARM MCUs that have the ARM MPU (not the NXP) are affected.

**Comment by MaureenHelm** [comment]
This test passes on mimxrt1050_evk, which has the ARM MPU

**Comment by andrewboie** [comment]
@erwango @nashif if you have boards can you try to run this one with assertions enabled?
Some of our SOCs define way too many static MPU regions at boot, leaving little or none left behind for runtime programming. So it may be trying to program the MPU and failing to get a free region.

**Comment by andrewboie** [comment]
This is basically an instance of #6896

**Comment by andrewboie** [comment]
per @nashif  #7893 fixes this

**Comment by erwango** [comment]
Fixed by #7893
